make-release.sh

```diff
@@ -27,132 +67,185 @@ Usage: $0 --version [VERSION TO RELEASE]
 Example: $0 --version 7.25.2 -t
 
 Options: 
-  --trigger-release, -t  tag this release
-  --no-commit, -n        do not commit changes to branches
+  --trigger-release, -t   Tag this release
+  --nocommit, -n          Do not commit changes to git branches
+  --nopush                Do not push changes to git remote
+  -tmp, --use-tmp-dir     Use a fresh git clone in a temporary directory
 "
 }
 
-set -e
-
 if [[ ! ${VERSION} ]]; then
   usage
   exit 1
 fi
 
-if [[ ${USE_TMP_DIR} -eq 1 ]]; then
-  cd /tmp/ && tmpdir=tmp-${0##*/}-$VERSION && git clone $REPO $tmpdir && cd /tmp/$tmpdir
-fi
+gitClone() {
+  if [[ ${USE_TMP_DIR} -eq 1 ]]; then
+    # Getting a local copy with all ta
+    cd /tmp/
+    tmpdir=tmp-${0##*/}-$VERSION
+    rm -rf "$tmpdir"
+    git clone $REPO "$tmpdir"
+    cd "/tmp/$tmpdir"
+    git pull
```

Comment:
do we need to pull if we've just cloned?

Reply:
To face the situation when a merge occured during clone. (It happened during development, because I was using the clone wait time to finish reviews...)